webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultCsvValidationService.java

```diff
@@ -195,50 +231,72 @@ private Integer getAsInt(final CSVRecord row, final String header) {
+            validationResults.add(CsvValidationResult.builder()
                     .row(recordNumber)
                     .message("Record does not contain a group name")
+                    .isOk(false)
                     .build());
         }
 
         if (schoolYear == null) {
-            validationFailures.add(CsvValidationFailure.builder()
+            validationResults.add(CsvValidationResult.builder()
                     .row(recordNumber)
                     .message("Record does not contain a school year")
+                    .isOk(false)
                     .build());
         }
     }
 
-    private void validateSchool(final List<CsvValidationFailure> validationFailures,
-                                                      final Set<String> availableSchools,
+    private void validateSchool(final List<CsvValidationResult> validationResults,
+                                                      final List<Organization> availableSchools,
                                                       final Set<String> completedSchools,
                                                       final String school,
                                                       final long recordNumber) {
-        if (!availableSchools.contains(school)) {
-            validationFailures.add(CsvValidationFailure.builder()
+        if (availableSchools.stream().noneMatch(x -> Objects.equals(x.getNaturalId(), school))) {
+            validationResults.add(CsvValidationResult.builder()
                     .row(recordNumber)
                     .message("School [" + school + "] can not be found")
+                    .isOk(false)
                     .build());
         }
         if (completedSchools.contains(school)) {
-            validationFailures.add(CsvValidationFailure.builder()
+            validationResults.add(CsvValidationResult.builder()
                     .row(recordNumber)
                     .message("School [" + school + "] is out-of-order: sort CSV by school, group, year")
+                    .isOk(false)
                     .build());
         }
     }
 
+    private static CsvValidationResult newSuccessValidationResult(final long row, final int groupCount, final Organization school) {
+        return CsvValidationResult
+                .builder()
+                .row(row)
+                .message(String.format(ValidationSuccessFormat, groupCount, groupCount == 1 ? "": "s", school.getName(), school.getNaturalId()))
+                .isOk(true)
+                .build();
+    }
+
+    private static boolean allOk(final List<CsvValidationResult> validationResults) {
+        return validationResults.stream().allMatch(CsvValidationResult::isOk);
+    }
+
+    private static Organization getSchoolOrganization(final List<Organization> availableSchools, final String naturalId) {
+       return availableSchools.stream().filter(x -> Objects.equals(x.getNaturalId(), naturalId)).findFirst().get();
```

Comment:
This `.get()` should be safe in that it will only get called in case all validations are successes, one of those validations being the school natural id.